plugins/modules/iam_user.py

```diff
@@ -205,8 +306,23 @@ def create_or_update_user(connection, module):
             changed = True
         except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
             module.fail_json_aws(e, msg="Unable to create user")
+
+        # Wait for user to be fully available before continuing
+        if module.params.get('wait'):
+            wait_iam_exists(connection, module)
+
+        if module.params.get('password') is not None:
+            create_or_update_login_profile(connection, module)
     else:
-        changed = update_user_tags(connection, module, params, user)
+        login_profile_result = None
+        update_result = update_user_tags(connection, module, params, user)
+
+        if module.params['update_password'] == "always" and module.params.get('password') is not None:
+            login_profile_result = create_or_update_login_profile(connection, module)
+        elif module.params.get('remove_password') is not None:
```

Comment:
```suggestion
        elif module.params.get('remove_password'):
```

Currently, setting `remove_password=False` will also delete the password :)